### PR TITLE
Don't use static overloads on instance for QuickInfo

### LINKED ISSUE #13462: Wrong number of overloads in QuickInfo
**Version Used**: 
VS2015, Update 3
**Steps to Reproduce**:

```
public class Program
 { 
    static void Main(string[] args) 
    {
        new T().M$$();
    }
}
public class T
{
    public void M() { }
    public static void M(int i) { }
}
```

When I hoover over the M in the main method a quickinfo with + 1 overload is shown.

**Expected Behavior**:
image
**Actual Behavior**:
image

### PR DESCRIPTION
**Pull request by antonssonj**
Fix #13462

### CODE REVIEW

**Comment by jmarolf** [comment]
test vsi please

**Comment by antonssonj** [comment]
I can't see the details of the failed test. Can someone with access give me the details?

**Comment by antonssonj** [comment]
Retest vsi please

File: src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs [hunk 1]
```diff
@@ -4181,6 +4181,36 @@ public class Methods2
             await VerifyWithReferenceWorkerAsync(markup, MainDescription(expectedDescription));
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs]: I see that many of the other tests in this file also have `Completion` here, but it should have `QuickInfo`, as the test below it does.

Also, can you add a `WorkItem` attribute linking it to the reported issue?

File: src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs [hunk 2]
```diff
@@ -276,8 +276,10 @@ internal abstract partial class AbstractSemanticQuickInfoProvider : AbstractQuic
             var symbols = semanticModel.GetSymbols(token, document.Project.Solution.Workspace, bindLiteralsToUnderlyingType: true, cancellationToken: cancellationToken);
 
             var bindableParent = document.GetLanguageService<ISyntaxFactsService>().GetBindableParent(token);
-            var overloads = semanticModel.GetMemberGroup(bindableParent, cancellationToken);
-
+            var parentSymbol = semanticModel.GetSymbolInfo(bindableParent).Symbol;
+            var overloads = semanticModel.GetMemberGroup(bindableParent, cancellationToken)
+                                         .Where(s => parentSymbol == null || enclosingType.InheritsFromOrEquals(parentSymbol.ContainingType) || s.IsStatic == parentSymbol.IsStatic);
+
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs]: Not trying to make a statement about the correctness of this approach, but how similar is it to the code in completion which already handles static/instance?

**Comment by Pilchie** [comment]
Thanks for the submission.  Tagging the rest of @dotnet/roslyn-ide for review.

**Comment by Pilchie** [comment]
BTW - does VB suffer a similar issue?  If so, do you want to fix it at the same time?  If not, want to add a test for VB to keep it from regressing in the future?

**Comment by antonssonj** [comment]
VB have the same problem, but the fix fix the problem for VB to. I have added a test for VB.

**Comment by Pilchie** [comment]
@dotnet-bot retest microbuild_prtest please

@dotnet/roslyn-ide can someone else take a look at this please?

**Comment by CyrusNajmabadi** [comment]
I had the same question as @Pilchie

File: src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs (hunk 2 above)
Comment (antonssonj) [inline src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs]: That's a good question,  I looked around (took more time then the fix) and I think this is the place were the completion code handles this: https://github.com/dotnet/roslyn/blob/master/src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs#L541 

The structur of the code for completion is quite different and I can't find any good help method to reuse. I also want to keep the change as small as possible.

By the way, I think I found another small bug when I tried the completion logic. 
```
public class Program
 { 
    static void Main(string[] args) 
    {
        new T().M($$);
    }
}
public class T
{
    public void M() { }
    public static void M(int i) { }
}
```
Press ctrl+space at the $$. In the list that is shown (why does it include that much?) there is the parameter from the static method 'i:'

**Comment by antonssonj** [comment]
Retest windows_release_unit32_prtest please, retest  windows_release_unit64_prtest please

**Comment by antonssonj** [comment]
Retest windows_release_unit32_prtest please

**Comment by antonssonj** [comment]
Retest windows_release_unit32_prtest please

**Comment by antonssonj** [comment]
Retest windows_release_unit32_prtest please

**Comment by antonssonj** [comment]
@Pilchie I tried to answer your question, do you have time to take a look at this?

**Comment by jaredpar** [comment]
We apologize, but we are closing this PR due to code drift. We regret letting this PR go so long without attention, but at this point the code base has changed too much for us to revisit older PRs. Going forward, we will manage PRs better under an SLA currently under draft in issue #26266 – we encourage you to add comments to that draft as you see fit. Although that SLA is still in draft form, we nevertheless want to move forward with the identification of older PRs to give contributors as much early notice as possible to review and update them.

If you are interested in pursuing this PR, please reset it against the head of master and we will reopen it. Thank you!

**Pull request closed without merging**